README.md

```diff
@@ -1225,6 +1225,102 @@ You should set `keyStore` and `trustStore` before starting the members. See the
+```
+
+Validation is done by .NET and delegated to OS and you need to make sure your server certificate is trusted by your OS.
+Please refer to [this blog](https://blogs.msdn.microsoft.com/webdev/2017/11/29/configuring-https-in-asp-net-core-across-different-platforms/) for how to configure your OS to trust your server certificates.
+
+##### Certificate name validation
+
+Server certificate CN or SAN field can be validated against a valid you set into configuration. This option is disabled by default.
+
+Example usage:
+
+```c#
+sslConfig.SetProperty(SSLConfig.ValidateCertificateName, "true");
+sslConfig.SetProperty(SSLConfig.CertificateName, "hazelcast.org");
+```
+
+##### TLS/SSL Protocol
+
+You can configure the TLS/SSL protocol using `SSLConfig.SslProtocol` option. Valid options are string value of `System.Security.Authentication.SslProtocols` Enum. 
+Depending on your .Net Framework/Net core version below values are valid:
+
+* **None**    : Allows the operating system to choose the best protocol to use. 
+* **Ssl2**    : SSL 2.0 Protocol. *RFC 6176 prohibits the usage of SSL 2.0.* 
+* **Ssl3**    : SSL 3.0 Protocol. *RFC 7568 prohibits the usage of SSL 3.0.*
+* **Tls**     : TLS 1.0 Protocol described in RFC 2246
+* **Tls11**   : TLS 1.1 Protocol described in RFC 4346 (.Net Framework 4.5+)
+* **Tls12**   : TLS 1.2 Protocol described in RFC 5246 (.Net Framework 4.5+)
+
+
+#### 6.1.3. Mutual Authentication
+
+As explained above, Hazelcast members have key stores used to identify themselves (to other members) and Hazelcast clients have trust stores used to define which members they can trust.
+
+Using mutual authentication, the clients also have their key stores and members have their trust stores so that the members can know which clients they can trust.
+
+To enable mutual authentication, firstly, you need to set the following property on the server side in the `hazelcast.xml` file:
+
+```xml
+<network>
+    <ssl enabled="true">
+        <properties>
+            <property name="javax.net.ssl.mutualAuthentication">REQUIRED</property>
+        </properties>
+    </ssl>
+</network>
+```
+
+You can see the details of setting mutual authentication on the server side in the [Mutual Authentication section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#mutual-authentication) of the Hazelcast IMDG Reference Manual.
+
+On the client side, you have to provide client certificate and its password if there is one. Here is how you do it:
+
+```c#
+sslConfig.SetProperty(SSLConfig.CertificateFilePath, "client pfx file path");
+sslConfig.SetProperty(SSLConfig.CertificatePassword, "client pfx password");
+```
+
+The provided certificate file should be a PFX file that has private and public keys.The file path should be set with `SSLConfig.CertificateFilePath`.
+If you choose to set a password to it, you need to provide it configuration with `SSLConfig.CertificatePassword` option.
```

Comment:
... you need to provide it to the configuration using the `SSLConfig.CertificatePassword` option.